path_data.go

```diff
@@ -252,6 +252,12 @@ func (b *versionedKVBackend) pathDataWrite() framework.OperationFunc {
 			return logical.ErrorResponse("missing path"), nil
 		}
 
+		isTrimmed := false
+		if req.Operation == logical.CreateOperation && strings.HasSuffix(key, " ") {
+			key = strings.TrimSpace(key)
```

Comment:
`strings.TrimSpace` actually trims both leading and trailing space. If that's what you want then you're good, but if you just want trailing space you'll want to use `strings.TrimRight`.

Reply:
And if you do want to trim leading spaces too, then probably makes sense to update the warning below accordingly.